src/AgentVisibilityContext.tsx

```diff
@@ -37,16 +47,54 @@ export const VisibilityProvider = ({ children }: { children: ReactNode }) => {
     setHiddenAgents(noAgentsSelectedMap);
   };
 
-  const handleVisibilityCheckboxChange = (agentName: string) => {
-    setHiddenAgents((prevHiddenAgents) =>
-      getNewSelectionMap(agentName, prevHiddenAgents)
-    );
+  const getChildren = (agentName: string): string[] => {
+    const agentEntry = uiDisplayData.find((entry) => entry.name === agentName);
+    if (agentEntry === undefined) {
+      return [];
+    }
+    return agentEntry.displayStates.map((state) => state.name);
   };
 
-  const handleHightlightCheckboxChange = (agentName: string) => {
-    setHighlightedAgents((prevHighlightedAgents) =>
-      getNewSelectionMap(agentName, prevHighlightedAgents)
-    );
+  const handleVisibilityCheckboxChange = (
+    agentName: string,
+    childAgentName?: string
```

Comment:
sorry to be so picky about this, but this also doesn't really make sense. the children are different states the agent can be in. It's not another agent name. So this second thing only is defined if you click on a child checkbox right? I might reverse this, have the first argument just be the name of the checkbox, and have an optional `parentName` if it's a child checkbox.